src/brevitas/quant/experimental/float_base.py

```diff
@@ -50,13 +53,38 @@ def exponent_bias(exponent_bit_width):
         return 2 ** (exponent_bit_width - 1) - 1
 
 
-class Fp8e4m3Mixin(ExponentBiasMixin):
+class MaxFloatInfNaNMixin(ExtendedInjector):
+
+    @value
+    def max_value(
+            exponent_bit_width, mantissa_bit_width, exponent_bias, nan_values, inf_values,
+            saturating):
+        return get_max_value(
+            exponent_bit_width,
+            mantissa_bit_width,
+            exponent_bias,
+            nan_values,
+            inf_values,
+            saturating)
+
+
+class Fp8e4m3Mixin(ExponentBiasMixin, MaxFloatInfNaNMixin):
```

Comment:
We need to create 4 more quantizers, two for weights, two for activations (e4m3 weights, e5m2 weights, etc.)
You can just inherit from them and add proxy_class I think

Reply:
Fp8e4m3Weight(FloatWeightBase):
    bit_width = 8
    exponent_bit_width = 4
    mantissa_bit_width = 3
    nan_values = (('111',))
    inf_values = None
    saturating = True